src/main/java/org/variantsync/diffdetective/diff/difftree/DiffNode.java

```diff
@@ -1069,6 +1063,11 @@ public void assertSemanticConsistency() {
                 Assert.assertTrue(afterParent.isIf() || afterParent.isElif(), "After parent " + afterParent + " of " + this + " is neither IF nor ELIF!");
             }
         }
+
+        // If and elif nodes have a formula
+        if (this.isIf() || this.isElif()) {
+            Assert.assertTrue(this.getDirectFeatureMapping() != null, "If or elif without feature mapping!");
```

Comment:
Maybe it's a good idea to also check for the absence of `featureMapping` in other node types?

Reply:
It is.

Reply:
Implemented.